nimbus-core/src/main/java/com/antheminc/oss/nimbus/channel/web/WebActionController.java

```diff
@@ -190,10 +211,40 @@ else if(o.getValue() instanceof Param) {
 		return ResponseEntity.status(HttpStatus.OK).body(null);
 	}
 	
+	@RequestMapping(value=URI_PATTERN_P+"/event/upload",consumes = MediaType.MULTIPART_FORM_DATA_VALUE,  produces="application/json", method=RequestMethod.POST)
+	public Object handleUpload(HttpServletRequest req, @RequestParam("pfu") MultipartFile file, @RequestParam("domain") String domain) {
+		if (file.isEmpty()) {
```

Comment:
Please review other methods in this controller class and see if there is a way to avoid writing logic specific to handling the file other than just recognizing the type of request and delegating to dispatcher to do the rest.

Reply:
@sandeepmantha can you look at this one?

Reply:
We will bring back the `FileImporterGateway` and let the framework clients have access and use this bean.  In doing so, they can create their own controller and wire up the logic themselves.

Reply:
Fixed.